rust/lance-index/src/vector/pq/storage.rs

```diff
@@ -413,11 +413,12 @@ impl ProductQuantizationStorage {
         writer: &mut FileWriter<ManifestDescribing>,
     ) -> Result<usize> {
         let batch_size: usize = 10240; // TODO: make it configurable
-        for i in (0..self.batch.num_rows()).step_by(batch_size) {
-            let offset = i * batch_size;
+        let mut offset = 0;
+        while offset < self.batch.num_rows() {
```

Comment:
is this the same as `for offset in (0..self.batch_num_rows().step_by(batch_size)` ,and just don't do `let offset = i * batch_size`?